treeherder/etl/perf.py

```diff
@@ -150,7 +150,8 @@ def load_perf_artifacts(project_name, reference_data, job_data, datum):
             summary_properties.update(extra_summary_properties)
             summary_signature_hash = _get_signature_hash(
                 summary_properties)
-
+            subtests = PerformanceSignature.objects.filter(
+                signature_hash__in=subtest_signatures)
```

Comment:
I see the problem here, you're pulling out all signatures whose hash matches the subtest signatures list, but that will include signatures across multiple repositories. This is what's causes the duplicates, I think.

The fix would be to also filter on repository here, but I realized there's a better way of implementing this that will avoid that problem altogether. Will follow up in the bug.